### PR TITLE
 Use unique_ptr for all opaque types

### PR DESCRIPTION
**Pull request by ahayzen-kdab**

### CODE REVIEW

File: cxx-qt-gen/src/extract.rs [hunk 1]
```diff
@@ -517,7 +517,14 @@ fn extract_invokable(
     let return_is_opaque = if let Some(return_type) = &return_type {
         matches!(
```
Comment (LeonMatthesKDAB) [inline cxx-qt-gen/src/extract.rs:518]: couldn't this use `return_type.qt_type.is_opaque()`?
This feels like code duplication of QtType::is_opaque() :thinking:

File: cxx-qt-gen/test_outputs/basic_only_invokable_return.cpp [hunk 2]
```diff
@@ -24,14 +24,14 @@ QString
 MyObject::helloMessage(const QString& msg)
 {
   const std::lock_guard<std::mutex> guard(m_rustObjMutex);
-  return rustStringToQString(m_rustObj->helloMessage(msg));
+  return std::move(*m_rustObj->helloMessageWrapper(msg));
```
Comment (LeonMatthesKDAB) [inline cxx-qt-gen/test_outputs/basic_only_invokable_return.cpp:27]: this feels a lot nicer :+1:

File: cxx-qt-lib/src/qstring.rs [hunk 3]
```diff
@@ -42,60 +43,7 @@ pub struct QString {
     _pinned: PhantomData<PhantomPinned>,
 }
 
-// TODO: figure out how to make Syntax and Example compile as code
```
Comment (LeonMatthesKDAB) [inline cxx-qt-lib/src/qstring.rs:45]: This whole file cleaned up nicely :)

File: cxx-qt-lib/src/qvariant.rs [hunk 4]
```diff
@@ -99,167 +95,76 @@ pub struct QVariant {
...
-    type Id = type_id!("QVariant");
-    type Kind = cxx::kind::Opaque;
+    #[doc(hidden)]
+    unsafe fn __drop(mut repr: MaybeUninit<*mut c_void>) {
+        unique_ptr_qvariant_drop(&mut repr)
+    }
 }
 
-pub enum VariantImpl {
+pub enum VariantEnum {
```
Comment (LeonMatthesKDAB) [inline cxx-qt-lib/src/qvariant.rs:167]: I'd prefer this be called "VariantValue" or something similar.

File: cxx-qt-lib/src/qvariant.rs [hunk 5]
```diff
@@ -7,19 +7,16 @@
...
-    mem::{self, ManuallyDrop, MaybeUninit},
-    ops::{Deref, DerefMut},
-    pin::Pin,
+    mem::MaybeUninit,
 };
 
 #[repr(u8)]
 #[allow(dead_code)]
-enum QVariantType {
+enum QVariantEnum {
```
Comment (LeonMatthesKDAB) [inline cxx-qt-lib/src/qvariant.rs:19]: I think we should stick with the "QVariantType" name here.
Having an enum named "Enum" doesn't really add value to the name

Comment (ahayzen-kdab) [reply cxx-qt-lib/src/qvariant.rs:19]: Done

File: cxx-qt-gen/src/extract.rs (hunk 1 above)
Comment (ahayzen-kdab) [reply cxx-qt-gen/src/extract.rs:518]: Done

**Pull request merged**